### ISSUE TITLE
React Select Menu uses render prop "open"

### ISSUE THREAD

**Issue by JSin** [tailwindlabs/tailwindui-issues#1135]
**What component (if applicable)**
- URL for category: https://tailwindui.com/components/application-ui/forms/select-menus
- Component name: All of them

**Describe the bug**
Since `headlessui v1.2.0`, the render prop for select menu is no longer required (https://github.com/tailwindlabs/headlessui/pull/466). The example components don't use the render prop `open` in a way that seems like it needs to exist in the example.

For consistency, it seems to be we should pick one or the other (including in the [headlessui docs](https://headlessui.dev/react/listbox#listbox)) to avoid confusion.

**Comment by JSin** [comment]
I realized there are actually more cases as the need for `open` was also removed from `Disclosure`, `Menu`, and `Popover` components.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your suggestion!
Much appreciated! 🙏

The `open` prop is not removed and is still present in the Headless UI codebase. It can still be useful to use this information to swap out classes or icons based on the `open` value.

It is true that we made it simpler for the communication between the Transition and other components and requires less bootstrap-y code.

That said, we kept the code in the Tailwind UI codebase for backwards compatibility unless we are sure that people are also using the newer Headless UI versions.

I'm sure we will update them eventually and I took note of it internally now that we also have new components that require a version bump either way. 

Thanks for bringing this to our attention!
